test/test_bundle_importer.js

```diff
@@ -0,0 +1,189 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2012-2015 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+describe('Bundle Importer', function() {
+  var addCharms, bundleImporter, BundleImporter, getById, loadCharm,
+      notifications, utils;
+
+  before(function(done) {
+    YUI().use('bundle-importer', 'juju-tests-utils', function(Y) {
+      BundleImporter = Y.juju.BundleImporter;
+      utils = Y['juju-tests'].utils;
+      done();
+    });
+  });
+
+  beforeEach(function() {
+    notifications = {
+      add: utils.makeStubFunction()
+    };
+    loadCharm = utils.makeStubFunction();
+    addCharms = utils.makeStubFunction();
+    getById = utils.makeStubFunction();
+    bundleImporter = new BundleImporter({
+      env: {},
+      db: {
+        notifications: notifications,
+        charms: {
+          add: addCharms,
+          getById: getById
+        }
+      },
+      fakebackend: {
+        _loadCharm: loadCharm
+      }
+    });
+  });
+
+  afterEach(function() {
```

Comment:
Unfortunately in order to use the cleanups we need defined before/afterEach methods.